enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -50,53 +50,33 @@ status: implementable
 >driver even be allowed to initiate killing a Pod?  If so, what is required and what approach should be employed?
-> 2. CVO vs. OLM for delivery.  May need @derekwaynecarr to cast the deciding vote.  Latest pro-con list below in the
->implementation details install section.  Top level question to answer is do we want this feature installed on
->all OCP clusters, or have it be an optional component.  If optional, OLM is the defacto answer.  If it always needs to
->be there, does that mandate CVO, or as some reviewers have speculated, *perhaps not in the near future*.
 
 ## Summary
 
-This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based
-means for OpenShift cluster administrators to share `Secrets` / `ConfigMaps` defined in one namespace with
-other namespaces.
-
-Any owners/authors/users of Pods in those namespaces can opt into the consumption of the content of those shared `Secrets` /
- `ConfigMaps` during their Pod's execution.
-
-This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
-**EDITORIAL NOTE:** if/when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
-https://github.com/openshift/enhancements.  Or if that PR is replaced by another proposal, or this proposal, this proposal will
-point that out.
+This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based means for OpenShift cluster administrators to share `Secrets` / `ConfigMaps` defined in one namespace with other namespaces.
 
+Any owners/authors/users of Pods in those namespaces can opt into the consumption of the content of those shared `Secrets` / `ConfigMaps` during their Pod's execution.
 
 ## Motivation
 
 Simplify provision and consumption of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces.
+There are numerous use cases where information should be defined only once on the cluster and shared broadly, such as:
+
+- Distribution of custom PKI certificate authorities, such as corporate self-signed certificates.
+- Simple Content Access certificates used to access RHEL subscription content.
+  See also the [Subscription Content Access](/enhancements/subscription-content/subscription-content/access.md) proposal.
 
 ### Goals
 
 - Provide easy to use sharing of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces.
-
-- When a shared `Secret` or `ConfigMap` changes, the container filesystem content the CSI driver injects into the Pod
-needs to reflect those changes.
-
-- When a cluster-admin revokes access to shared `Secrets` or `ConfigMaps` that was previously granted, that data should
-no longer be accessible.
+- When a shared `Secret` or `ConfigMap` changes, the container filesystem content the CSI driver injects into the Pod needs to reflect those changes.
```

Comment:
Kubernetes did the mistake that Secrets and ConfigMap are automatically refreshed. With hundreds of thousands of pods in a cluster, watching and refreshing the ConfigMaps and Secrets takes significant amount of resources. Later on, Kubernetes introduces opt-out from updating content of these volumes.

IMO it would make sense either to start with opt-in for the refresh of the volumes (but it might create confusion) or follow Kubernetes and have opt-out.

Reply:
I'm good with adding that feature as well.

@adambkaplan if you agree as well, can you add?

Reply:
Would it make sense for us to have a separate controller that labels Secrets and ConfigMaps that are shared? That way the watches of the CSI driver daemons only act on secrets that have a specific label key, whose value maps to the appropriate share instance. Unclear if this would reduce the volume of watch events sent to the driver.

This separate controller introduces its own set of issues, such as leader election. Unlike the storage driver, we only want one active controller that labels Secrets and ConfigMaps based on which resources have been shared.

Reply:
Since we are already concerned with scaling, I would prefer we allow opting in for refreshes. For the Build use case we won't need entitlements to be refreshed.

Reply:
sure thing .... so aside from any updates to this EP, either add this to our existing "productization" jira or create a new jira to get this added